src/tribler/core/components/gigachannel_manager/tests/test_gigachannel_manager.py

```diff
@@ -113,32 +111,40 @@ def test_updated_my_channel(personal_channel, gigachannel_manager, tmpdir):
     gigachannel_manager.download_manager.start_download.assert_called_once()
 
 
-async def test_check_and_regen_personal_channel_torrent(personal_channel, gigachannel_manager):
+async def test_check_and_regen_personal_channel_torrent_wait(personal_channel, gigachannel_manager):
     with db_session:
         chan_pk, chan_id = personal_channel.public_key, personal_channel.id_
-        chan_download = MagicMock()
 
-        async def mock_wait(*_):
-            pass
+    # Test wait for status OK
+    await gigachannel_manager.check_and_regen_personal_channel_torrent(
+        channel_pk=chan_pk,
+        channel_id=chan_id,
+        channel_download=MagicMock(wait_for_status=AsyncMock()),
+        timeout=0.5
+    )
 
-        chan_download.wait_for_status = mock_wait
-        # Test wait for status OK
-        await gigachannel_manager.check_and_regen_personal_channel_torrent(chan_pk, chan_id, chan_download, timeout=0.5)
 
-        async def mock_wait_2(*_):
-            await asyncio.sleep(3)
+async def test_check_and_regen_personal_channel_torrent_sleep(personal_channel, gigachannel_manager):
+    with db_session:
+        chan_pk, chan_id = personal_channel.public_key, personal_channel.id_
 
-        chan_download.wait_for_status = mock_wait_2
-        # Test timeout waiting for seeding state and then regen
+    async def mock_wait(*_):
+        await asyncio.sleep(3)
 
-        f = MagicMock()
+    f = MagicMock()
 
-        async def mock_regen(*_):
-            f()
+    async def mock_regen(*_):
+        f()
 
-        gigachannel_manager.regenerate_channel_torrent = mock_regen
-        await gigachannel_manager.check_and_regen_personal_channel_torrent(chan_pk, chan_id, chan_download, timeout=0.5)
-        f.assert_called_once()
+    with patch.object(GigaChannelManager, 'regenerate_channel_torrent', mock_regen):
```

Comment:
I think, the code can be simplified a bit (with AsyncMock, a separate `f` mock is not necessary):
```suggestion
    mock_regen = AsyncMock()
    with patch.object(GigaChannelManager, 'regenerate_channel_torrent', mock_regen):
        ...
    mock_regen.assert_called_once()
```

Reply:
Fixed